cobbler/tftpgen.py

```diff
@@ -1138,15 +1138,18 @@ def build_kernel_options(
         # This also is architecture specific. E.g: Some ARM consoles need: console=ttyAMAx,BAUDRATE
         # I guess we need a serial_kernel_dev = param, that can be set to "ttyAMA" if needed.
         if system and arch == Archs.X86_64:
-            if system.serial_device or system.serial_baud_rate:
-                if system.serial_device:
-                    serial_device = system.serial_device
-                else:
+            if (
+                system.serial_device > -1
+                or system.serial_baud_rate != enums.BaudRates.DISABLED
+            ):
+                if system.serial_device == -1:
                     serial_device = 0
-                if system.serial_baud_rate:
-                    serial_baud_rate = system.serial_baud_rate.value
                 else:
+                    serial_device = system.serial_device
+                if system.serial_baud_rate == enums.BaudRates.DISABLED:
```

Comment:
This should never be true since above you checked that this is NOT the case?

Reply:
The above check is an `or`, not an `and`. It's possible for `system_serial_device` to be > -1 while `system.serial_baud_rate`== enums.BaudRates.DISABLED.

Reply:
At least for me, the patch did fix the issue and I was able to boot an AlmaLinux 8.6 Anaconda install which displayed its progress on the screen (whereas previously it would not).

Reply:
Oh boy it really seems that I was distracted while writing this. Sorry for the noise.